backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,71 @@
+import { RedisClient } from 'redis';
+import { CachingObject } from '../data/CachingObject';
+import { Promotion } from '../entity/Promotion';
+
+export class CachingService {
+  private client: RedisClient;
+
+  constructor(client: RedisClient) {
+    this.client = client;
+  }
+
+  async cacheLatLonValues(
+    placeId: string,
+    latitude: number,
+    longitude: number
+  ): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      const value: CachingObject = { lat: latitude, lon: longitude };
+      this.client.setex(
+        placeId,
+        2592000,
+        JSON.stringify(value),
+        (err: Error | null, data: string) => {
+          if (err) {
+            return reject(err);
+          }
+          return resolve(true);
+        }
+      );
+    });
+  }
+
+  async getLatLonValue(placeId: string): Promise<CachingObject> {
+    return new Promise((resolve, reject) => {
+      this.client.get(placeId, (err: Error | null, data: string | null) => {
+        if (err) {
+          // todo: https://github.com/ubclaunchpad/foodies/issues/89
+          reject(err);
+        }
+
+        if (data) {
+          return resolve(JSON.parse(data));
+        } else {
+          return resolve(JSON.parse(''));
+        }
```

Comment:
Very minor, but I think this can be simplified to:

```suggestion
        return resolve(JSON.parse(data ?? ''));
```

If you're expecting the second case to execute only when data is not `null` or `undefined`!

Reply:
Ooo, thank you! :))